Section 1: Introduction and Setup/4-RoR Development Environment/testing_project/app/channels/application_cable/channel.rb

```diff
@@ -0,0 +1,4 @@
+module ApplicationCable
```

Comment:
what is this file for?

Reply:
`channel.rb` inherits from ActionCable::Channel::Base and its responsible for pushing contents to the subscribers of the channel.

Reply:
This kind of answer shows me that you jstu copy paste what you read without understanding it.

How is it responsible to push content? Where does it does the pushing? how does it do the pushing? magically?

Reply:
When `consumer` subscribes to a channel, they are called the `subscriber`. Action Cable uses Pub/Sub or Publish-Subscribe approach to communicate between server and clients. In a Pub/Sub system, the publisher send contents to the channel and those subscribers that subscribed to that specific channel would receive the the contents from the publisher. Some examples of a pub/sub usage is to deliver contents to users' feed based on the topic they subscribed in a content delivering system or in a chat system where users in a same room would be in a same channel.

Reply:
you are now describing what Pub/Sub & Action Cable is. You still haven't explained what is this file for and how it works.

Reply:
Rails by default creats a parent `ApplicationCable::Channel` class for encapsulating shared logic between your channels which is the `channels/application_cable/channel.rb` file.
You can then create additional files in `/channels` for example `chat.rb` as below for different channels for consumers to subscribe to.
```
# /application_cable/chat.rb

class Chat < ApplicationCable::Channel
end
```